general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -227,6 +227,38 @@ def get_balance(self) -> dict[str, float]:
         """
         return self.get_request("/balance")
 
+    @overload
+    def get_user_info(self, query: None = None) -> dict[str, object]: ...
+
+    @overload
+    def get_user_info(self, query: Mapping[str, object]) -> dict[str, dict[str, object]]: ...
+
+    def get_user_info(
+        self, query: Mapping[str, object] | None = None
+    ) -> dict[str, object] | dict[str, dict[str, object]]:
+        """Gets a dictionary of the user's info.
+
+        .. note::
+
+            SUPERTECH users can submit an optional :code:`query` argument which can be used
+            to search for the info of arbitrary users on the server.
+
+        Args:
+            query: An optional dictionary of query parameters to use to search for user records.
+                Is only available to SUPERTECH users. Defaults to None.
+        """
+        if query is None:
+            user_info = self.get_request("/get_user_info", query={})
+            try:
+                return list(user_info.values())[0]
+            except IndexError:
+                raise gss.SuperstaqServerException(
+                    "Something has gone wrong, the server has returned an empty response."
+                )
```

Comment:
can this actually occur?

if so, let's just check if the dict is empty directly instead of using a try..except, e.g.:
```python
if not user_info:
    raise...
return ...
```

Reply:
It shouldn't occur, but I would rather catch the error if it did